### PR TITLE
Provide operator configuration via FLINK_PROPERTIES fixes #135

### LINKED ISSUE #135: Replace environment variable for configuration injection
Environment variable `OPERATOR_FLINK_CONFIG` should be replaced with `FLINK_PROPERTIES` to support the stock Flink Docker images.

https://github.com/docker-flink/docker-flink/pull/82

We should also update the example to use that image.

### PR DESCRIPTION
**Pull request by tweise**
The old environment variable is retained for a transition period. The docker-flink containers now support environment variable substitution as well, so we can get rid of the custom entry point!

### CODE REVIEW

File: pkg/controller/flink/task_manager_controller_test.go [hunk 1]
```diff
@@ -136,7 +136,7 @@ func TestTaskManagerHACreateSuccess(t *testing.T) {
 			"high-availability.cluster-id: app-name-"+hash+"\n"+
 			"taskmanager.host: $HOST_IP\n",
 			common.GetEnvVar(deployment.Spec.Template.Spec.Containers[0].Env,
-				"OPERATOR_FLINK_CONFIG").Value)
+				"FLINK_PROPERTIES").Value)
```
Comment (glaksh100) [inline pkg/controller/flink/task_manager_controller_test.go]: Shall we continue to test for `OPERATOR_FLINK_CONFIG` being populated here?

**Comment by anandswaminathan** [comment]
Just to understand correctly, 

We are changing from `OPERATOR_FLINK_CONFIG ` to `FLINK_PROPERTIES`, and  keeping `OPERATOR_FLINK_CONFIG` for now for backward compatibility. Right ?

**Comment by tweise** [comment]
@anandswaminathan yes, we can remove `OPERATOR_FLINK_CONFIG` after a couple of releases to allow users to migrate their Docker files.

File: pkg/controller/flink/task_manager_controller_test.go (hunk 1 above)
Comment (tweise) [reply pkg/controller/flink/task_manager_controller_test.go]: Added the check.

**Comment by glaksh100** [comment]
This change will cause applications to restart, right (since the hash has changed)?

**Comment by tweise** [comment]
@glaksh100 thanks for pointing this out, a noted it in the commit message.

**Pull request merged**